moveit_commander/src/moveit_commander/planning_scene_interface.py

```diff
@@ -306,14 +239,59 @@ def __make_cylinder(name, pose, height, radius):
         return co
 
     @staticmethod
-    def __make_planning_scene_diff_req(collision_object, attach=False):
+    def __make_plane(name, pose, normal=(0, 0, 1), offset=0):
+        co = CollisionObject()
+        co.operation = CollisionObject.ADD
+        co.id = name
+        co.header = pose.header
+        p = Plane()
+        p.coef = list(normal)
+        p.coef.append(offset)
+        co.planes = [p]
+        co.plane_poses = [pose.pose]
+        return co
+
+    @staticmethod
+    def __make_planning_scene_diff_req(collision_object):
         scene = PlanningScene()
         scene.is_diff = True
         scene.robot_state.is_diff = True
-        if attach:
+        if isinstance(collision_object, AttachedCollisionObject):
             scene.robot_state.attached_collision_objects = [collision_object]
         else:
             scene.world.collision_objects = [collision_object]
         planning_scene_diff_req = ApplyPlanningSceneRequest()
         planning_scene_diff_req.scene = scene
         return planning_scene_diff_req
+
+
+def __create_add_attach_methods(shape):
+    """Add methods add_xxx() and attach_xxx() to class PlanningSceneInterface for a given shape"""
+    make_method = getattr(PlanningSceneInterface, '_PlanningSceneInterface__make_' + shape)
+
+    def add(self, name, pose, *args, **kwargs):
+        co = make_method(name, pose, *args, **kwargs)
+        self._PlanningSceneInterface__submit(co)
+
+    def attach(self, link, name, pose=None, *args, **kwargs):
+        touch_links = kwargs.pop('touch_links', [link])
+        aco = AttachedCollisionObject()
+        if pose is not None:
+            try:
+                aco.object = make_method(name, pose, *args, **kwargs)
+            except TypeError:  # thrown if touch_links was specified as an unnamed last argument
```

Comment:
@v4hn Is that what you are thinking of?
```suggestion
            except TypeError:  # thrown if touch_links was specified as an unnamed last argument
                               # (not matching the signature of make_method, which doesn't expect this additional argument)
```

Reply:
Something like that, yes.